### DISCUSSION TITLE
Get collection in websocket event

### DISCUSSION THREAD

**Discussion by Zyles** [directus/directus#19276] [Feature Requests]
### Summary

Currently when you subscribe to an event you get something like:

```
message {
  type: 'subscription',
  event: 'init',
  data: [
    {
      ...
    }
  ]
}

or:

message {
  type: 'subscription',
  event: 'create',
  data: [
    {
      ...
    }
  ]
}
```

But it is not clear what collection we are actually referring to.

I need a way to know what collection this event is about.

### Basic Example

Add collection to the event.

```
message {
  type: 'subscription',
  event: 'create',
  collection: 'tasks',
  data: [
    {
      ...
    }
  ]
}
```

### Motivation

Be able to act on events and route it internally in application to the correct event handler.

### Detailed Design

Donno.

### Requirements List

Donno.

### Drawbacks

No drawbacks what I can see.

### Alternatives

None.

### Adoption Strategy

Not breaking change.

**Comment by br41nslug** [comment]
It definitely could help in certain cases :) you should however already know (or be able to trace) what collection you've subscribed to based on the `uid` of your subscription. Judging from the code snippets you're not actively using but it is meant to be able to match the subscription events to your original subscription even if lets say you have 2 subscriptions to the same collection with different queries.

**Reply by Zyles** [reply]
I saw the uid and sure that is one way to do it.

But when I need only the collection name to process the event it would be better to get it right away in the message. Otherwise I need to do a lookup in a database on my end on the subscription to see what collection it belongs to.

Having collection in the message would save me from doing that in this scenario.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
